app/_components/PoiCard.tsx

```diff
@@ -86,7 +89,8 @@ export function PoiCard({
           body: JSON.stringify(data),
         }
       );
-      const JSONresponse = response.json() as Promise<Response>;
+      const JSONresponse = (await response.json()) as Promise<Response>;
+      setCheckLevel(true);
       return JSONresponse;
     } catch (error) {
       console.error(error);
```

Comment:
Should extract the setState to the inactive state to the finally, if something error between the `try` block, the state will not be set back to true again.

```suggestion
      const JSONresponse = (await response.json()) as Promise<Response>;
      return JSONresponse;
    } catch (error) {
      console.error(error);
    } finally {
      setCheckLevel(true);
```